lib/bap/bap_project.ml

```diff
@@ -330,7 +319,12 @@ let create_exn
       let disasm = Disasm.create g in
       let program = MVar.read program in
       let program =
-        symbolize_synthetic program (Disasm.insns disasm) spec in
+        if MVar.is_updated spec then
```

Comment:
why are you injecting here, instead of putting the code in a plugin and using an external injection point? Are there any invariants on which the core code now depends and which are provided by this injector?